### ISSUE TITLE
Returned LogoutId and Session ID was null when logged in with external authentication

### ISSUE THREAD

**Issue by nareshkathi** [IdentityServer/IdentityServer4#2775]
returned LogoutId and Session ID was null when logged in with Facebook.

I am trying use Facebook external authentication using Identity Server 4. Successfully able to login with Facebook and got the Claims as below
![image](https://user-images.githubusercontent.com/40530509/47956358-8241b980-df60-11e8-90e0-33c1f7a9acf6.png)
![image](https://user-images.githubusercontent.com/40530509/47956374-b321ee80-df60-11e8-8037-2be5472b8e77.png)

Above are the Claims that i got, and i don't see any LogoutId or session id. and hence i tried to create LogoutId by calling  IIdentityServerInteractionService.CreateLogoutContextAsync() , but even then, the returned LogoutId was NULL, thus external logout was not working.

Please suggest.

**Comment by brockallen** [comment]
> returned LogoutId and Session ID was null when logged in with Facebook.

These will only be present if the upstream provider is OIDC.

**Comment by nareshkathi** [comment]
Thanks for response.
But if they are not needed, how to make sure that i/user logged out of Facebook when i/user clicks on LogOut button ?
When i click on LogOut, i was successfully able to logged out of local Identityserver session but not from Facebook. 

I see this issue with Google also.

Thanks.

**Comment by brockallen** [comment]
I don't think either of those providers support signout.

**Comment by nareshkathi** [comment]
So, is there a way to end session from external authentication providers like  Facebook/Google using identity server 4 ? can we use iframes ?

**Comment by brockallen** [comment]
> So, is there a way to end session of external authentication providers like Facebook/Google using identity server 4

You misunderstand -- your question has nothing to do with IdentityServer. Your question is about Google and facebook and ASP.NET Core's support for them.

**Comment by nareshkathi** [comment]
Sorry, let me rephrase the question. I am using the combination of IdentityServer 4 and ASP .Net Identity in my application to maintain authentication and authorization. In this case, we implemented Google/Facebook external authentication and not able to perform external logout.  is there any way to logout of external authentication providers(like Google/Facebook) session ?

Thanks

**Comment by brockallen** [comment]
>  is there any way to logout of external authentication providers(like Google/Facebook) session

As I said, I think they don't support logout. Think about it -- it's not in the _business model_ to support logout.

**Comment by nareshkathi** [comment]
got it. 

Thank you Allen, i really appreciate your prompt response.
